src/pyinfra/facts/server.py

```diff
@@ -52,12 +52,28 @@ def command(self):
 
 class TmpDir(FactBase):
     """
-    Returns the temporary directory of the current server, if configured.
+    Returns the temporary directory of the current server.
+
+    According to POSIX standards, checks environment variables in this order:
+    1. TMPDIR (if set and accessible)
+    2. TMP (if set and accessible)
+    3. TEMP (if set and accessible)
+    4. Falls back to /tmp
     """
 
     @override
     def command(self):
-        return "echo $TMPDIR"
+        return """
+if [ -n "$TMPDIR" ] && [ -d "$TMPDIR" ] && [ -w "$TMPDIR" ]; then
+    echo "$TMPDIR"
+elif [ -n "$TMP" ] && [ -d "$TMP" ] && [ -w "$TMP" ]; then
+    echo "$TMP"
+elif [ -n "$TEMP" ] && [ -d "$TEMP" ] && [ -w "$TEMP" ]; then
+    echo "$TEMP"
+else
+    echo "/tmp"
```

Comment:
We shouldn't provide a default here, the fallback is specified at the config level currently (https://github.com/pyinfra-dev/pyinfra/blob/3.x/src/pyinfra/api/config.py#L29). 

To be honest I think this way is nicer, but we can't break the config for backwards compatibility so if no matching env vars the fact still needs to return a blank string.